### PR TITLE
make @timed return NamedTuple (# 34147)

### LINKED ISSUE #34147: Return named tuple from `@timed`
Can we return a named tuple from `@timed` macro?
```julia
julia> value, time, bytes, gctime, gcdiff = @timed rand(10^6);
```
is error prone and hard to remember.

### PR DESCRIPTION
**Pull request by jw3126**
fix #34147

### CODE REVIEW

File: base/util.jl [hunk 1]
```diff
@@ -276,21 +276,21 @@ See also [`@time`](@ref), [`@timev`](@ref), [`@elapsed`](@ref), and
 [`@allocated`](@ref).
 
 ```julia-repl
-julia> val, t, bytes, gctime, memallocs = @timed rand(10^6);
+julia> o = @timed rand(10^6);
```
Comment (StefanKarpinski) [inline base/util.jl]: Maybe call this `t` instead? Calling it `o` is a bit odd.

File: base/util.jl [hunk 2]
```diff
@@ -276,21 +276,21 @@ See also [`@time`](@ref), [`@timev`](@ref), [`@elapsed`](@ref), and
...
-julia> bytes
+julia> o.bytes
 8000256
 
-julia> gctime
+julia> o.gctime
 0.0055765
 
-julia> fieldnames(typeof(memallocs))
+julia> fieldnames(typeof(o.gcdiff))
```
Comment (JeffBezanson) [inline base/util.jl]: How about `gcstats` as the name of this field?

Comment (StefanKarpinski) [reply base/util.jl]: Right, the rainfall about using a named tuple is that now the names are significant.

**Comment by jw3126** [comment]
I think the CI fail is unrelated.

**Comment by StefanKarpinski** [comment]
Would it be sacrelidge to name these `gc_time` and `gc_stats`?

**Comment by JeffBezanson** [comment]
Yes :stuck_out_tongue:

**Comment by jw3126** [comment]
Can somebody rerun CI?

**Comment by jw3126** [comment]
Merge?

**Comment by StefanKarpinski** [comment]
Seems good but I think everyone is taking a break until the new year.

File: base/util.jl [hunk 3]
```diff
@@ -276,23 +276,26 @@ See also [`@time`](@ref), [`@timev`](@ref), [`@elapsed`](@ref), and
...
-julia> fieldnames(typeof(memallocs))
+julia> propertynames(stats.gcstats)
 (:allocd, :malloc, :realloc, :poolalloc, :bigalloc, :freecall, :total_time, :pause, :full_sweep)
 
-julia> memallocs.total_time
+julia> stats.gcstats.total_time
 5576500
 ```
+
+!!! compat "Julia 1.4"
```
Comment (JeffBezanson) [inline base/util.jl]:
```suggestion
!!! compat "Julia 1.5"
```

File: base/util.jl [hunk 4]
```diff
@@ -276,23 +276,26 @@ See also [`@time`](@ref), [`@timev`](@ref), [`@elapsed`](@ref), and
...
+    The return type of this macro was changed from `Tuple` to `NamedTuple` in Julia 1.4
```
Comment (JeffBezanson) [inline base/util.jl]:
```suggestion
    The return type of this macro was changed from `Tuple` to `NamedTuple` in Julia 1.5.
```

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**